requests-for-comments/2024-07-01-alerting.md

```diff
@@ -0,0 +1,160 @@
+### Request for comments (RFC): Threshold-based alerting
+
+#### Introduction
+
+This RFC proposes an architecture for an alerting feature in PostHog, which allows users to define conditions that
+trigger alerts based on their data. The system will evaluate these conditions at regular intervals and notify users
+when conditions are met. The design includes detailed logging for debugging and auditing, and efficient handling of
+task execution using Celery.
+
+#### Goals
+
+1. **Condition Evaluation**: Define and evaluate alert conditions at specified intervals.
+2. **Notification System**: Notify users when alert conditions are met.
+3. **Logging and Auditing**: Store detailed logs of each alert check.
+4. **Efficiency and Scalability**: Handle tasks efficiently to reduce load on the system.
+
+#### Architecture
+
+1. **Models**:
+    - **Alert**: Represents the alert configuration.
+    - **AlertCheck**: Logs each evaluation of an alert, including results and notification statuses.
+
+2. **Task Execution**:
+    - Use Celery to handle periodic checks and notifications.
```

Comment:
just checking whether it should be temporal rather than celery (i don't even know how we'd choose 🙈)